src/header/common/link.rs

```diff
@@ -0,0 +1,1142 @@
+    Start,
+    /// stylesheet.
+    Stylesheet,
+    /// subsection.
+    Subsection,
+    /// successor-version.
+    SuccessorVersion,
+    /// up.
+    Up,
+    /// versionHistory.
+    VersionHistory,
+    /// via.
+    Via,
+    /// working-copy.
+    WorkingCopy,
+    /// working-copy-of.
+    WorkingCopyOf,
+    /// ext-rel-type.
+    ExtRelType(String)
+}
+
+////////////////////////////////////////////////////////////////////////////////
+// Struct methods
+////////////////////////////////////////////////////////////////////////////////
+
+impl Link {
+    /// Create `Link` from a `Vec<LinkValue>`.
+    pub fn new(link_values: Vec<LinkValue>) -> Link {
+        Link { values: link_values }
+    }
+
+    /// Get the `Link` header's `LinkValue`s.
+    pub fn values(&self) -> &[LinkValue] {
+        self.values.as_ref()
+    }
+
+    /// Add a `LinkValue` instance to the `Link` header's values.
+    pub fn push_value(&mut self, link_value: LinkValue) {
+        self.values.push(link_value);
+    }
+}
+
+impl LinkValue {
+    /// Create `LinkValue` from URI-Reference.
+    pub fn new<T>(uri: T) -> LinkValue
+        where T: Into<Cow<'static, str>> {
+        LinkValue {
+            link: uri.into(),
+            rel: None,
+            anchor: None,
+            rev: None,
+            href_lang: None,
+            media_desc: None,
+            title: None,
+            title_star: None,
+            media_type: None,
+            link_extension: None,
+        }
+    }
+
+    /// Get the `LinkValue`'s value.
+    pub fn link(&self) -> &str {
+        self.link.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `rel` parameter(s).
+    pub fn rel(&self) -> Option<&[RelationType]> {
+        self.rel.as_ref().map(AsRef::as_ref)
+    }
+
+    /// Get the `LinkValue`'s `anchor` parameter.
+    pub fn anchor(&self) -> Option<&str> {
+        self.anchor.as_ref().map(AsRef::as_ref)
+    }
+
+    /// Get the `LinkValue`'s `rev` parameter(s).
+    pub fn rev(&self) -> Option<&[RelationType]> {
+        self.rev.as_ref().map(AsRef::as_ref)
+    }
+
+    /// Get the `LinkValue`'s `hreflang` parameter(s).
+    pub fn href_lang(&self) -> Option<&[LanguageTag]> {
+        self.href_lang.as_ref().map(AsRef::as_ref)
+    }
+
+    /// Get the `LinkValue`'s `media` parameter(s).
+    pub fn media_desc(&self) -> Option<&[MediaDesc]> {
+        self.media_desc.as_ref().map(AsRef::as_ref)
+    }
+
+    /// Get the `LinkValue`'s `title` parameter.
+    pub fn title(&self) -> Option<&str> {
+        self.title.as_ref().map(AsRef::as_ref)
+    }
+
+    /// Get the `LinkValue`'s `title*` parameter.
+    pub fn title_star(&self) -> Option<&str> {
+        self.title_star.as_ref().map(AsRef::as_ref)
+    }
+
+    /// Get the `LinkValue`'s `type` parameter.
+    pub fn media_type(&self) -> Option<&Mime> {
+        self.media_type.as_ref()
+    }
+
+    /// Get the `LinkValue`'s `link-extension` parameter.
+    pub fn link_extension(&self) -> Option<&str> {
```

Comment:
As pointed out in https://github.com/hyperium/hyper/issues/1085, this is incorrect. A later commit simply removed the `link_extension` pieces, allowing for them to be added correctly later.